inc/dashboard/provider.class.php

```diff
@@ -413,6 +416,317 @@ static function nbTicketsGeneric(
    }
 
 
+   public static function nbTicketsByAgreementStatusAndTechnician(
+      string $case = "",
+      array $params = []
+   ):array {
+      $DBread = DBConnection::getReadConnection();
+
+      $default_params = [
+         'label'         => "",
+         'icon'          => 'far fa-clock',
+         'apply_filters' => [],
+      ];
+      $params = array_merge($default_params, $params);
+
+      $query_criteria  = [];
+      $search_criteria = [];
+
+      $laTable = Sla::getTable();
+      $laFk    = Sla::getForeignKeyField();
+      if ($case == 'Ola') {
+         $laTable = Ola::getTable();
+         $laFk    = Ola::getForeignKeyField();
+      }
+      $table = Ticket::getTable();
+      $laFk .=  '_ttr';
+      $ticketUserTable = Ticket_User::getTable();
+      $userTable = User::getTable();
+
+      $ownExceeded = Ticket::generateSLAOLAComputation('time_to_own', $table);
+      $resolveExceeded = Ticket::generateSLAOLAComputation('time_to_resolve', $table);
+      $slaState = "IF ($ownExceeded AND $resolveExceeded, 3, IF ($resolveExceeded, 2, IF ($ownExceeded, 1, 0)))";
+      $config = Config::getConfigurationValues('core');
+      if ($config['names_format'] == User::FIRSTNAME_BEFORE) {
+         $first = "firstname";
+         $second = "realname";
+      } else {
+         $first = "realname";
+         $second = "firstname";
+      }
+
+      $friendlyName = "CONCAT(`$userTable`.$first, ' ', `$userTable`.$second)";
+      $query_criteria = [
+         'COUNT' => 'cpt',
+         'SELECT'    => [
+            new QueryExpression("$friendlyName as `username`"),
+            "$userTable.name",
+            new QueryExpression("$slaState as `sla_state`"),
+         ],
+         'FROM'   => $table,
+         'INNER JOIN' => [
+            "$ticketUserTable as gu" => [
+               'FKEY' => [
+                  'gu' => 'tickets_id',
+                  $table => 'id',
+                  [
+                     'AND' => [
+                        "gu.type" => Ticket_User::ASSIGN
+                     ]
+                  ]
+               ],
+            ],
+            $userTable => [
+               'FKEY' => [
+                  $userTable => 'id',
+                  'gu' => 'users_id',
+               ],
+            ],
+         ],
+         'WHERE'  => [
+            "$table.is_deleted" => 0,
+         ] + getEntitiesRestrictCriteria($table),
+         'GROUPBY' => [
+            'sla_state',
+            "$userTable.id",
+         ],
+         'ORDER' => [
+            'sla_state DESC',
+            'cpt DESC',
+            "$userTable.id",
+         ]
+      ];
+
+      unset($params['apply_filters']['group_tech']);
```

Comment:
this card is not compatible with group_tech filter. To prevent false results, it is necessary to not apply the filter.

I did not find a proper way to do this. Comments welcome